lib/src/outputs/multi_output.dart

```diff
@@ -0,0 +1,50 @@
+import 'package:logger/src/log_output.dart';
+import 'package:logger/src/logger.dart';
+
+/// Logs simultaneously to multiple [LogOutput] outputs.
+class MultiOutput extends LogOutput {
+  List<LogOutput> _outputs;
+
+  MultiOutput(List<LogOutput> outputs) {
+    _outputs = _normalizeOutputs(outputs);
+  }
+
+  /// Returns [true] if contains [output].
+  bool containsOutput(LogOutput output) {
+    for (var o in _outputs) {
+      if (identical(o, output)) return true;
+
+      if (o is MultiOutput) {
+        if (o.containsOutput(output)) return true;
+      }
+    }
+    return false;
+  }
+
+  List<LogOutput> _normalizeOutputs(List<LogOutput> outputs) {
+    if (outputs == null || outputs.isEmpty) return [];
+
+    var outputs2 = List<LogOutput>.from(outputs);
+
+    // Remove null or self referenced output:
+    outputs2.removeWhere((o) =>
+        o == null ||
+        identical(o, this) ||
+        (o is MultiOutput && o.containsOutput(this)));
+
+    return outputs2;
+  }
```

Comment:
I agree that if the output is null this is an engineer error. But the main idea is to avoid `LogOutput` self references.

The method `containsOutput` checks deeply if the output `o` makes a self reference to `this`, avoiding an infinity loop.

In general, Log instances and outputs shouldn't be fixed in the code, this is why `LoggerFactory` exists. The idea is to be able to change the log output using external properties (or other ways) when wanted/needed. So, is not only about engineer errors, but ensure that it works in odd situations.

Reply:
I'm not sure I follow what you mean by "In general, Log instances and outputs shouldn't be fixed in the code, this is why LoggerFactory exists.". But I fully support the idea that outputs as used in this class could be added programmatically. 
Would you not consider it a bug in the calling code when it adds a self referencing output? Or similarly a longer cycle (A -> B -> C -> B for example).